packages/components/dashboard/inventory-card/inventory-card.tsx

```diff
@@ -0,0 +1,388 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { HealthBody } from '@openshift-console/dynamic-plugin-sdk-internal';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import {
+  Gallery,
+  GalleryItem,
+  Card,
+  CardBody,
+  CardHeader,
+  CardTitle,
+  Skeleton,
+} from '@patternfly/react-core';
+import {
+  BucketClassPhaseMap,
+  NamespaceStorePhaseMap,
+  ObjectBucketClaimPhaseMap,
+  PhaseType,
+} from '../../../constants';
+import {
+  NooBaaBucketClassModel,
+  NooBaaNamespaceStoreModel,
+  NooBaaObjectBucketClaimModel,
+} from '../../../models';
+import {
+  BucketClassKind,
+  K8sResourceKind,
+  NamespaceStoreKind,
+} from '../../../types';
+import { referenceForModel } from '../../../utils';
+import {
+  RedExclamationCircleIcon,
+  BlueInProgressIcon,
+} from '../../../utils/status/icons';
+import {
+  BucketClassPopOver,
+  DataResourcesPopOver,
+  OBCPopOver,
+} from '../../resource-pages/ResourcePopOver';
+import {
+  bucketClassResource,
+  nameSpaceStoreResource,
+  bucketClaimResource,
+} from '../../resources';
+import './inventory-card.scss';
+
+const OBC_LIST_PATH =
+  '/k8s/all-namespaces/' + referenceForModel(NooBaaObjectBucketClaimModel);
+const BUCKET_CLASS_LIST_PATH =
+  '/mcgms/cluster/resource/' + referenceForModel(NooBaaBucketClassModel);
+const DATA_RESOURCE_LIST_PATH =
+  '/mcgms/cluster/resource/' + referenceForModel(NooBaaNamespaceStoreModel);
+
+type obcType = { name: string; ns: string };
+type displayItems = {
+  statusMap: { [key: string]: number };
+  errorList: string[] | obcType[];
+  processingList: string[] | obcType[];
+};
+
+type PopupProps = {
+  type: PhaseType;
+  statusmap: { [key: string]: number };
+  data: string[] | obcType[];
+};
+
+type CustomGalleryItemProps = {
+  listItems;
+  listLoaded: boolean;
+  listError;
+  redirectPath: string;
+  resource: string;
+  statusMap: { [key: string]: number };
+  errorPopup: React.ReactNode;
+  processingPopup: React.ReactNode;
+};
```

Comment:
check if it is possible to reduce number of props for this component, else we will re factor this component further later (not now)...

Reply:
cool will explore